sdk/search/azure-search-documents/azure/search/documents/_service/_utils.py

```diff
@@ -151,3 +159,28 @@ def listize_synonyms(synonym_map):
     # type: (dict) -> dict
     synonym_map["synonyms"] = synonym_map["synonyms"].split("\n")
     return synonym_map
+
+def get_access_conditions(model, match_condition=MatchConditions.Unconditionally):
+    # type: (Any, MatchConditions) -> Tuple[Dict[int, Any], AccessCondition]
+    error_map = {
+        401: ClientAuthenticationError,
+        404: ResourceNotFoundError
+    }
+
+    if not hasattr(model, 'e_tag'):
+        return (error_map, None)
+
+    try:
+        if_match = prep_if_match(model.e_tag, match_condition)
+        if_none_match = prep_if_none_match(model.e_tag, match_condition)
+        if match_condition == MatchConditions.IfNotModified:
+            error_map[412] = ResourceModifiedError
+        if match_condition == MatchConditions.IfModified:
```

Comment:
While somewhat unusual (and possibly nonsensical), you could get a 412 response here as well if the HTTP method wasn't GET.

Reply:
will change it to 412

Reply:
Not change - add. Both 304 (which you'd get for a GET) and 412 (which you'd get for other methods) should result in the same exception type.

```suggestion
        if match_condition == MatchConditions.IfModified:
            error_map[304] = ResourceNotModifiedError
```